lazy_dataset/core.py

```diff
@@ -1132,6 +1132,90 @@ def apply(self, apply_fn: callable) -> 'Dataset':
         else:
             return apply_fn(self)
 
+    def cache(
+            self,
+            lazy: bool = True,
+            keep_mem_free: Tuple[Union[float, int], str] = None,
+            catch_filter_exception: Any = False
```

Comment:
Does `dataset.catch().cache(lazy=False)` do the same as `dataset.cache(lazy=False, catch_filter_exception=True)`?
If yes, I would drop `catch_filter_exception` in the signature.

Reply:
Ideally, yes. But unfortunately `dataset.catch()` is not indexable. If we somehow can give datasets keys without requiring them to be indexable, both would be identical.

Reply:
This is related to the point of "Allow datasets that are not indexable if lazy=False" from above. If we allow this, both are identical and we an remove `catch_fitler_exception`

Reply:
I think (as mentioned below) we need a flag like `ordered` to figure out if the input is random or not.